src/main/java/ay2122s1_cs2103t_w16_2/btbb/model/order/Deadline.java

```diff
@@ -0,0 +1,95 @@
+package ay2122s1_cs2103t_w16_2.btbb.model.order;
+
+import static ay2122s1_cs2103t_w16_2.btbb.commons.util.AppUtil.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents the deadline of an order in BTBB.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Deadline {
+    private static final String INPUT_FORMAT = "dd-MM-yyyy HHmm";
+    public static final DateTimeFormatter INPUT_DATETIME_FORMATTER = DateTimeFormatter.ofPattern(INPUT_FORMAT);
+    private static final String OUTPUT_FORMAT = "MMM d yyyy h:mm a";
+    private static final DateTimeFormatter OUTPUT_DATETIME_FORMATTER = DateTimeFormatter.ofPattern(OUTPUT_FORMAT);
+    public static final String MESSAGE_CONSTRAINTS = "Deadline should be in "
+            + INPUT_FORMAT + " format and should occur in the future.";
+
+    private final LocalDateTime deadline;
+
+    /**
+     * Constructs a {@code Deadline}.
+     *
+     * @param deadline A valid deadline.
+     */
+    public Deadline(String deadline) {
+        requireNonNull(deadline);
+        checkArgument(isValidDeadline(deadline), MESSAGE_CONSTRAINTS);
+        this.deadline = LocalDateTime.parse(deadline, INPUT_DATETIME_FORMATTER);
+    }
+
+    /**
+     * Returns true if a given string is a valid deadline.
+     *
+     * @param test String input to check.
+     * @return True if the deadline is valid. False otherwise.
+     */
+    public static boolean isValidDeadline(String test) {
+        if (test == null) {
+            return false;
+        }
+
+        try {
+            LocalDateTime dateTime = LocalDateTime.parse(test, INPUT_DATETIME_FORMATTER);
+            return dateTime.isAfter(LocalDateTime.now());
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+    }
```

Comment:
We need a different `isValidStorageDeadline` for storage deadlines. Since the dates in storage can be in the past.